### ISSUE TITLE
JAX-RS request context modified after client request

### ISSUE THREAD

**Issue by sdaschner** [OpenLiberty/open-liberty#6317]
The request context (incl. information) which is injected into a JAX-RS resource is modified after a client request has been made. This impacts the UriInfo builder which produces different URIs (and probably other aspects as well).

Reproducer: https://github.com/sdaschner/jaxrs-test/tree/request-context, tested with OL `18.0.0.3/wlp-1.0.22.cl180320180905-2337`

This might be related to #6273.

**Comment by WhiteCat22** [comment]
I've reproduced the issue with trace

**Comment by WhiteCat22** [comment]
Might have found the problem:
```
[2/19/19 16:27:06:726 CST] 00000042 id=489f61a6 om.ibm.ws.jaxrs20.injection.metadata.InjectionRuntimeContext > setRuntimeCtxObject Entry  
                                                                                                               javax.ws.rs.core.UriInfo
                                                                                                               org.apache.cxf.jaxrs.impl.UriInfoImpl@16270f05
```

looks like the client is creating a new UriInfoImpl and we are setting it.

**Comment by WhiteCat22** [comment]
Looking at this method in /com.ibm.ws.jaxrs.2.0.common/src/org/apache/cxf/jaxrs/utils/InjectionUtils.java
```
    // Liberty Change for CXF Begin
    @SuppressWarnings("unchecked")
    public static void injectManagedObjectContextField(AbstractResourceInfo cri,
                                                       Field f, Object o, Object value) {

        InjectionRuntimeContext irc = null;
        irc = InjectionRuntimeContextHelper.getRuntimeContext();
        irc.setRuntimeCtxObject(f.getType().getName(), value);
//      if (!cri.isSingleton()) {
//
//          irc.setRuntimeCtxObject(o.getClass().getName()
//                                  + "-" + f.getType().getName(), value);
//      } else {
//          ThreadLocalProxy<Object> proxy = (ThreadLocalProxy<Object>) cri.getContextFieldProxy(f);
//          if (proxy != null) {
//              proxy.set(value);
//          }
//
//      }
    }
```

**Comment by WhiteCat22** [comment]
I have found that subsequent requests behave correctly. It is only the first request that is problematic

**Comment by WhiteCat22** [comment]
[2/28/19 14:28:40:918 CST] 0000002b id=3b715636 org.apache.cxf.service.model.EndpointInfo                    > setAddress Entry  
                                                                                                               http://localhost:9080/foobar

This seems to be why we are getting the new address

**Comment by WhiteCat22** [comment]
I think org.apache.cxf.jaxrs.utils.HttpUtils.getEndpointAddress() should have some logic to get the address from the ThreadLocalUriInfo before getting it from the EndpointInfo, not sure what side effects that will have.

**Comment by WhiteCat22** [comment]
I think I found the problem. org.apache.cxf.service.model.EndpointInfo has 
```
private static ThreadLocal<EndpointReferenceType> threadLocal = new ThreadLocal<EndpointReferenceType>();
```
Which shouldn't be static, this is causing issues when we have multiple EndpointInfos

**Comment by WhiteCat22** [comment]
merged #6752
